pkg/test/environment.go

```diff
@@ -503,6 +507,25 @@ func (e *Environment) Context() context.Context {
 	return e.ctx
 }
 
+func (e *Environment) StartEmbeddedJetstream() (*nats.Conn, error) {
+	ports, err := freeport.GetFreePorts(1)
```

Comment:
Can you add an associated port in the Environment ports struct  & an environment variable for this port?
Also could you delete the hackier version of starting jetstream in the testenv : https://github.com/rancher/opni/blob/a49ecd410a51ebf95fa80025ff0bbabd63ff955e/pkg/test/environment.go#L650

Reply:
As discussed in Joe's comment I would prefer to not tie that work to the implementation of this feature.  I also think there is value in running a fatter Jetstream server for the test.Env as it also sets up nkey and other things that are used in integration tests.  The embedded server is a minimal server just for unit tests.